packages/deltas-websocket/src/client/client-impl.ts

```diff
@@ -168,33 +167,82 @@ export class LionWebClient {
             combinedFactoryFor(languageBases, commandSender),
             commandSender,
             lowLevelClient
-        ) // (need this constant non-inlined to set participationId later)
+        ) // (need this constant non-inlined for write-access to lastReceivedSequenceNumber and queryResolveById)
         return lionWebClient
     }
 
-    private enqueueQuery = async (queryRequest: QueryRequest): Promise<QueryResponse> =>
+    private readonly enqueueQuery = (queryRequest: QueryMessage): Promise<QueryMessage> =>
         new Promise((resolveResponse, rejectResponse) => {
-            this.queryPromiseById[queryRequest.queryId] = { resolveResponse, rejectResponse }
-            return this.lowLevelClient.sendMessage(queryRequest)
+            this.queryResolveById[queryRequest.queryId] = resolveResponse
+            this.lowLevelClient.sendMessage(queryRequest)
+                .catch(rejectResponse)
         })
 
-    signOn = async (queryId: LionWebId) =>
+    async subscribeToChangingPartitions(queryId: LionWebId, parameters: SubscribeToPartitionChangesParameters): Promise<void> {
         await this.enqueueQuery({
+            messageKind: "SubscribeToChangingPartitionsRequest",
+            queryId,
+            ...parameters,
+            protocolMessages: []
+        } as SubscribeToChangingPartitionsRequest)
```

Comment:
shouldn't this be `...Response`?

Reply:
No, this is the request part: it gets en-Q'd, meaning that the message is sent, and then the `resolve` callback of the associated `Promise` is stored so the promise can be resolved, so that queries can be `await`ed. Would this be useful as documentation of the `enqueueQuery` method>